.github/workflows/build.yml

```diff
@@ -23,9 +23,26 @@ on:
     types: [ opened, synchronize, reopened ]
 
 jobs:
-  test:
+  build-agent-and-model:
```

Comment:
I really like the way you split it. However, I read, in https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions#jobs, that 

> by default, jobs have no dependencies and run in parallel with each other

and I'm wondering how isolated it will be - i.e. if at the same time the runner would run build for scala 2.11 for the agent and also 2.13 for server - would there be a potential clash?

https://github.com/orgs/community/discussions/26769#discussioncomment-3253321

Reply:
My assumption is that this is the same `runs-on: ubuntu-latest` but probably it will be different instance of this OS. (But...I remember that there is some magic caching potentially happening...we'll see :D )